File: Objective-C/OCProjectDemo/OCProjectDemo/Moduls/Login/LoginViewController.m
//
//  LoginViewController.m
//  OCProjectDemo
//
//  Created by Mojy on 2018/3/9.
//  Copyright © 2018年 Mojy. All rights reserved.
//

#import "LoginViewController.h"

@interface LoginViewController ()

@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
@property (weak, nonatomic) IBOutlet UITextField *userNameTextField;
@property (weak, nonatomic) IBOutlet UITextField *passwordTextField;
@property (weak, nonatomic) IBOutlet UIButton *LoginButton;
@property (weak, nonatomic) IBOutlet UIButton *cancelButton;

@end

@implementation LoginViewController

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    [self.navigationController setNavigationBarHidden:YES animated:NO];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    [self initConfig];
    [self layoutConfig];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - UI
/**
 控件初始化设置
 */
- (void)initConfig
{
    self.view.backgroundColor = DefaultColorBlue;
    
    self.titleLabel.text = @"登录";
    self.titleLabel.font = Font(18);
    self.titleLabel.textColor = [UIColor whiteColor];
    

}

/**
 布局设置
 */
- (void)layoutConfig
{
    
}

#pragma mark - Handle

- (IBAction)Login:(id)sender
{
    
}

- (IBAction)cancel:(id)sender
{
    
}

@end

File: Swift/SwiftProjectDemo/SwiftProjectDemo/Moduls/Login/ViewController/LoginViewController.swift
//
//  LoginViewController.swift
//  SwiftDemo
//
//  Created by Mojy on 2018/1/30.
//  Copyright © 2018年 com.swiftdemo.app. All rights reserved.
//

import UIKit
import RxCocoa
import RxSwift

class LoginViewController: UIViewController {
    
    @IBOutlet weak var passwordTextField: UITextField!
    @IBOutlet weak var useranmeTextField: UITextField!
    
    @IBOutlet weak var usernameTitleLabel: UILabel!
    @IBOutlet weak var passwordTitleLabel: UILabel!
    
    @IBOutlet weak var loginButton: UIButton!
    
    var viewModel: LoginViewModel!
    
    let disopseBag = DisposeBag()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "用户登录"

        self.useranmeTextField.text = "18612345676"
        self.passwordTextField.text = "qweqwe123"

        // Do any additional setup after loading the view.
        viewModel = LoginViewModel.init(input: (useranmeTextField.rx.text.orEmpty.asDriver(), passwordTextField.rx.text.orEmpty.asDriver(),loginButton.rx.tap.asDriver()), service: ValidationService.instance)
        
        //bindings
//        viewModel.usernameUsable
//            .drive(onNext: {
//
//            })
//            .addDisposableTo(disopseBag)
        
        //
        viewModel.loginResult
            .drive(onNext: { [weak self] result in

                print("\(result)")
                self?.view.endEditing(true)

                switch result {
                case let .ok(message):
                    self?.showAlert(message: message)
                case .empty:
                    self?.showAlert(message: "")
                case let .failed(message):
                    self?.showAlert(message: message)
                }

            })
            .disposed(by: disopseBag)
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

extension LoginViewController {
    
    func showAlert(message: String) {
        let action = UIAlertAction(title: "确定", style: .default, handler: nil)
        let alertViewController = UIAlertController(title: nil, message: message, preferredStyle: .alert)
        alertViewController.addAction(action)
        present(alertViewController, animated: true, completion: nil)
    }
}
